components/journal/EditVisitForm.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+import supabaseClient from '@/lib/supabase/client';
+import { useJournalStore } from '@/lib/stores/journal';
+import { EditVisitFormProp, visitFormInputs } from '@/content/types';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { Button } from '../ui/button';
+import { X } from 'lucide-react';
+
+const EditVisitForm = ({ visitId, visitType, onClose }: EditVisitFormProp) => {
+  const editToiletVisit = useJournalStore((state) => state.editToiletVisit);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<visitFormInputs>({
+    defaultValues: {
+      type: visitType,
+    },
+  });
+
```

Comment:
Unclear why when I click edit for the first time it does not show default value marked on radio button, but if I do it again, then it does. I think somewhere there is a glitch between this information saved in database vs. state. @johnie

Reply:
You can also add toasters when user adds, removes and edit toilet visit, i forgot. <3